src/Nethermind/Nethermind.Network.Stats/NodeStatsLight.cs

```diff
@@ -113,9 +121,34 @@ public void AddNodeStatsDisconnectEvent(DisconnectType disconnectType, Disconnec
                 _lastRemoteDisconnect = disconnectReason;
             }
 
+            if (disconnectType == DisconnectType.Local)
+            {
+                if (_statsParameters.DelayDueToLocalDisconnect.TryGetValue(disconnectReason, out TimeSpan delay))
+                {
+                    UpdateDelayConnectDeadline(delay, NodeStatsEventType.LocalDisconnectDelay);
+                }
+            }
+            else if (disconnectType == DisconnectType.Remote)
+            {
+                if (_statsParameters.DelayDueToRemoteDisconnect.TryGetValue(disconnectReason, out TimeSpan delay))
+                {
+                    UpdateDelayConnectDeadline(delay, NodeStatsEventType.RemoteDisconnectDelay);
+                }
+            }
+
             Increment(NodeStatsEventType.Disconnect);
         }
 
+        private void UpdateDelayConnectDeadline(TimeSpan delay, NodeStatsEventType reason)
+        {
+            DateTime newDeadline = DateTime.Now + delay;
```

Comment:
DateTime.Now can have issues with daylight saving time? Would move to UtcTime or DateTimeOffset.